modem: ublox-sara-r4/u2 build error

**Describe the bug**
Modem ublox-sara-r4 driver will not compile.

**To Reproduce**
west build -b particle_boron

**Expected behavior**
Build and compile

**Impact**
Can not build and run with modem 

**Screenshots or console output**
```
drivers/modem/ublox-sara-r4.c:43:12: error: 'DT_INST_0_UBLOX_SARA_R4_MDM_POWER_GPIOS_CONTROLLER' undeclared here (not in a function); did you mean 'DT_INST_0_JEDEC_SPI_NOR_WP_GPIOS_CONTROLLER'?
  MODEM_PIN(DT_INST_0_UBLOX_SARA_R4_MDM_POWER_GPIOS_CONTROLLER,
```

**Additional context**
**commit a0fceff changed dts/bindings/modem/ublox,sara-r4.yaml** 
`compatible: "ubloc,sara-r4"`

needs to be reverted to:
`constraint: "ublox,sara-r4"`

--- a/dts/bindings/modem/ublox,sara-r4.yaml
+++ b/dts/bindings/modem/ublox,sara-r4.yaml
@@ -6,7 +6,7 @@ title: u-blox SARA-R4 modem
 description: >
     This is a representation of the u-blox SARA-R4 modem.
 
-compatible: "ubloc,sara-r4"
+compatible: "ublox,sara-r4"
 
 include: uart-device.yaml
 

